SU2_CFD/include/numerics/BFMInterpolator.hpp

```diff
@@ -0,0 +1,132 @@
+    * \param [in] rad_cell - Radial boundary coordinates of query cell.
+    * \returns Inclusion of query point into query cell.
+    */
+    bool RC_Inclusion(su2double axis, su2double radius, vector<su2double>ax_cell, vector<su2double>rad_cell);
+
+    /*!
+    * \brief Distance-weighted interpolation algorithm.
+    * \param [in] axis - Axial coordinate of query point.
+    * \param [in] radius - Radial coordinate of query point.
+    * \param [in] ax_cell - Axial boundary coordinates of query cell.
+    * \param [in] rad_cell - Radial boundary coordinates of query cell.
+    * \param [in] val_cell - Data points at cell boundary.
+    * \returns Interpolated data value at query point.
+    */
+    su2double DW_average(su2double axis, su2double radius, vector<su2double>ax_cell, vector<su2double>rad_cell, vector<su2double>val_cell);
+    
+    
+    su2double Vector_Dot_Product(vector<su2double>a, vector<su2double>b){
+        if(a.size() != b.size()){
+            SU2_MPI::Error("Input vectors are of unequal size", CURRENT_FUNCTION);
+        }
+        su2double dot_product{0};
+        for(size_t i=0; i<a.size(); ++i){
+            dot_product += a.at(i) * b.at(i);
+        }
+        return dot_product;
+    };
+
+    su2double Vector_Dot_Product(vector<su2double>a, su2double *b){
+        // if(a.size() != sizeof(b)/sizeof(b[0])){
+        //     SU2_MPI::Error("Input vectors are of unequal size", CURRENT_FUNCTION);
+        // }
+        su2double dot_product{0};
+        for(size_t i=0; i<a.size(); ++i){
+            dot_product += a.at(i)*b[i];
+        }
+        return dot_product;
+    };
+
+    su2double Vector_Dot_Product(su2double *b, vector<su2double> a){
+        // if(a.size() != sizeof(b)/sizeof(b[0])){
+        //     SU2_MPI::Error("Input vectors are of unequal size", CURRENT_FUNCTION);
+        // }
+        return Vector_Dot_Product(a, b);
+        // su2double dot_product{0};
+        // for(size_t i=0; i<a.size(); ++i){
+        //     dot_product += a.at(i)*b[i];
+        // }
+        // return dot_product;
+    };
+    
+    vector<su2double> Vector_Cross_Product(vector<su2double>a, vector<su2double>b){
+        vector<su2double> c;
+        if(a.size() != b.size()){
+            SU2_MPI::Error("Input vectors are of unequal size", CURRENT_FUNCTION);
+        }
+        if(a.size() > 2){
+            c.resize(a.size());
+            for(size_t i=0; i<a.size(); ++i){
+                c.at(i) = a.at((i+1) % a.size())*b.at((i+2)%b.size()) - a.at((i+2) % a.size())*b.at((i+1) % b.size());
+            }
+        }else{
+            c.resize(1);
+            c.at(0) = a.at(0)*b.at(1) - b.at(0)*a.at(1);
+        }
+        
+        return c;
+    };
+
+    public:
+    BFMInterpolator();
+    BFMInterpolator(ReadBFMInput *reader, CSolver *solver_container, CGeometry *geometry, CConfig *config);
+
+    void Interpolate(ReadBFMInput *reader, CSolver *solver_container, CGeometry *geometry);
+
+    ~BFMInterpolator();
```

Comment:
Const correctness and do not declare the default ctor and dtor if you are defaulting them (i.e. they are empty).

It is more important to document the public interface than the private methods of the class.